pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+)
+
+func newPackagePublishCmd() *cobra.Command {
+
+	var err error
+	var path string
+	var tag string
+
+	cmd := &cobra.Command{
+		Use:   "publish-sdk <package> <language>",
+		Args:  cobra.ExactArgs(2),
+		Short: "Publish a package SDK to supported package registries.",
+		Long:  "Publish a package SDK to supported package registries.",
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Validate the language
+
+			lang := args[1]
+
+			switch lang {
+			case "all":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "nodejs":
+				fmt.Println("Nodejs is supported, publishing")
+			case "python":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "java":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "dotnet":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			default:
+				return fmt.Errorf(
+					"language %s is not supported. Must be one of: \n"+
+						"- all \n"+
+						"- nodejs \n"+
+						"- python \n"+
+						"- java \n"+
+						"- dotnet ", lang,
+				)
+			}
+
+			return nil
+		},
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			pkg := args[0]
+			lang := args[1]
+
+			err = publish(pkg, lang, path, tag)
+			if err != nil {
+				return fmt.Errorf("error publishing: %s", err.Error())
+			}
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	cmd.PersistentFlags().StringVar(&tag, "tag", "",
+		`The tag for your NPM package, such as "latest""`)
+	return cmd
+}
+
+func publish(pkg, lang, path, tag string) error {
+
+	fmt.Printf("You are publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+	case "python":
+		publishToPyPI(pkg)
+	case "java":
+		publishToMaven(pkg)
+	case "dotnet":
+		publishToNuget(pkg)
+	default:
+		// Publish to all registries
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+		publishToPyPI(pkg)
+		publishToMaven(pkg)
+		publishToNuget(pkg)
+	}
+
+	return nil
+}
+
+func publishToNPM(pkg, path, tag string) error {
+	// this is likely where we need to go get the sdk info
+	// verify path
+	_, err := os.Stat(path)
+	if errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("could not find path to package directory: %s", path)
+	}
+
+	// Verify  npm exists set up: npm, user login
+	ex, err := executable.FindExecutable("npm")
+	if err != nil {
+		return fmt.Errorf("could not find npm: %w", err)
+	}
+
+	// TODO: Verify auth for npm
+
+	whoamiCmd := exec.Command(ex, "whoami")
+	whoami, err := whoamiCmd.Output()
+
+	if err != nil {
+		fmt.Print(string(whoami))
```

Comment:
Is this a leftover from testing?
If `whoamiCmd.Output()` failed, the `whoami` return value is meaningless and we should probably not print it.

Reply:
I find the output helpful:
```
npm ERR! code E401
npm ERR! 401 Unauthorized - GET https://registry.npmjs.org/-/whoami

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/guin/.npm/_logs/2023-05-10T22_51_33_692Z-debug.log
```
I'm happy to wrap it as an error as well.

Reply:
No, no having the output is good. Print to stdout was unexpected: I would have expected to print to stderr or use one of the Errorf log methods.

Running `npm whoami` locally, I see that it prints the error to stderr, though, not stdout. 

```
# redirect stdout to /dev/null still prints output
% npm whoami >/dev/null
npm ERR! code ENEEDAUTH
# ...

# redirect stderr to /dev/null prints no output
% npm whoami 2>/dev/null
[empty]
```

Since `whoamiCmd.Output()` only returns the stdout output, `string(whoami)` here will be empty when that happens, and this line won't actually print anything.

If we connect stderr of whoamiCmd to os.Stderr, we'll still get the error message printed properly:

```go
whoamiCmd.Stderr = os.Stderr
whoami, err := whoamiCmd.Output()
```